### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by s-idgunji**
Pull request

### CODE REVIEW

File: client_server/serverConfigParserHelper/serverConfigParser.cpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    }
+}
+
+ServerCommands getServerCommands(std::string fileName) {
+    _ServerCommands data;
+    ServerCommands commands;
+    getCommands(&data, parseSeverCommands, fileName);
+    commands.logFile += data.logFile;
+    commands.ntp += data.ntp;
+    commands.ptdStartCommand += data.ptdPath + " " + data.ptdOptions + " " + data.serialNumber;
```
Comment (xzfc) [inline client_server/serverConfigParserHelper/serverConfigParser.cpp:203]:
```suggestion
    commands.logFile = data.logFile;
    commands.ntp = data.ntp;
    commands.ptdStartCommand = data.ptdPath + " " + data.ptdOptions + " " + data.serialNumber;
```

File: client_server/maxValues/maxAmpsVoltsParser.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    (* testResults)[std::string(testName)] = maxValues;
+}
+
+std::map <std::string, MaxAmpsVolts> getMaxAmpsVolts(std::string fileName) {
+    std::map <std::string, MaxAmpsVolts> data;
+    getCommands(&data, parseMaxAmpsVolts, fileName);
+    std::map <std::string, MaxAmpsVolts> :: iterator it = data.begin();
+
+    for (; it != data.end(); it++) {
```
Comment (xzfc) [inline client_server/maxValues/maxAmpsVoltsParser.cpp:38]: Code style:

```suggestion
    for (auto &it : data) {
```

File: client_server/maxValues/maxAmpsVoltsParser.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    const char * testName = element->name->string;
+    json_value_s * testValue = element->value;
+    struct json_object_s* object = json_value_as_object(testValue);
+
+    //TODO:Add check
+    struct json_object_element_s* maxValuesElement = object->start;
+
+    while (maxValuesElement != nullptr) {
+        const char *elementNameString = maxValuesElement->name->string;
+        json_value_s * elementValue = maxValuesElement->value;
```
Comment (xzfc) [inline client_server/maxValues/maxAmpsVoltsParser.cpp:19]: Code style: please use consistent `*` placement in this file.
https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions

File: client_server/maxValues/maxAmpsVoltsParser.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// Created by julia on 01/12/2020.
+//
+
+#ifndef POWER_MAXAMPSVOLTSPARSER_H
+#define POWER_MAXAMPSVOLTSPARSER_H
+#include "./../clientServerParserLib/clientServerParserLib.h"
+#include <map>
+
+constexpr auto MAX_AMPS = "maxAmps";
+constexpr auto MAX_VOLTS = "maxVolts";
```
Comment (xzfc) [inline client_server/maxValues/maxAmpsVoltsParser.h:11]: Should be in `maxAmpsVoltsParser.h`.

File: client_server/clientServerParserLib/clientServerParserLib.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    } else {
+        struct json_string_s *command = json_value_as_string(element);
+        copyCommandWithCheck(command, commands);
+    }
+}
+
+void copyStringValueToDataField(struct json_value_s *element, std::string *command) {
+    if (element->type != json_type_string) {
+        std::cerr << "Wrong config file" << std::endl;
```
Comment (xzfc) [inline client_server/clientServerParserLib/clientServerParserLib.cpp:34]: Shouldn't we `exit()` here?

File: client_server/maxValues/maxAmpsVoltsParser.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        } else {
+            std::cout << "Wrong JSON key" << std::endl;
+            exit(0);
+        }
+        maxValuesElement = maxValuesElement->next;
+    }
+    (* testResults)[std::string(testName)] = maxValues;
+}
+
+std::map <std::string, MaxAmpsVolts> getMaxAmpsVolts(std::string fileName) {
```
Comment (xzfc) [inline client_server/maxValues/maxAmpsVoltsParser.cpp:33]: Code style:
```suggestion
std::map<std::string, MaxAmpsVolts> getMaxAmpsVolts(std::string fileName) {
```

**Review by xzfc — CHANGES_REQUESTED** [review]
General notes:

1. Move 3rd party dependencies like `cxxopts`, `json.h` into a separate directory so we could distinguish your code from the dependencies.

2. I suggest you move all source files into a single directory instead of a directory per each pair of files.

3. In the long run, it's better to fetch dependencies using git submodules or another dependency management tool (e.g. `ExternalProject` in CMake).

4. Please use consistent names for modules (e.g. `clientServerParserLib`, `interacting_with_process`). Use `camelCase` or `snake_case` everywhere.

5. Please follow this rule: https://google.github.io/styleguide/cppguide.html#Unnamed_Namespaces_and_Static_Variables
   I've commented on it in a few places, but there are more.
   In short, move internal declarations from `.h` to `.cpp`, and make them `static`.

**Review by xzfc — CHANGES_REQUESTED** [review]
